packages/lesswrong/components/comments/CommentsItem/CommentsItem.tsx

```diff
@@ -311,6 +400,45 @@ export const CommentsItem = ({
 
   const voteProps = useVote(comment, "Comments", votingSystem);
   const showInlineCancel = replyFormIsOpen && isMinimalist
+  const { create: createDoppelVote, data: newDoppelCommentVote } = useCreate({
+    collectionName: "DoppelCommentVotes",
+    fragmentName: "DoppelCommentVotesFragment",
+  })
+
+  const shouldWeDoppel = userHasDoppelComments(currentUser) // is LW/AF & is admin
+    && comment.doppelComments.length >= 2 // are there doppel comments to show?
+    && !(comment.ownDoppelCommentVote || newDoppelCommentVote) // has user already voted?
+
+  const potentialCommentsData = shouldWeDoppel
+    ? { potentialComments: potentialCommentBodies,
+        setActivePotentialComment, activePotentialComment,
+        doppelCommentVote: <DoppelCommentVote
+          className=''
+          comment={comment}
+          createDoppelVote={createDoppelVote}
+          potentialCommentBodies={potentialCommentBodies}
+          activePotentialComment={activePotentialComment}
+        />
+      }
+    : undefined
+  const replaceReplyButtonsWith = potentialCommentsData ? {
```

Comment:
I would pretty strongly prefer to refactor the `replyButton` logic to be declarative and all in one place, rather than overwriting whatever's in treeOptions, but only conditional on `potentialCommentsData`, and then having the default fallback...

```
let replyButton;

if (potentialCommentsData) {
  replyButton = <DoppelCommentVote
    className={classes.replyLink}
    comment={comment}
    createDoppelVote={createDoppelVote}
    potentialCommentBodies={potentialCommentBodies}
    activePotentialComment={activePotentialComment}
  />;
} else if (treeOptions.replaceReplyButtonsWith) {
 replyButton = treeOptions.replaceReplyButtonsWith(comment);
} else {
  replyButton = <a ...>;
}
```